### PR TITLE
Feat/reducer container

### PR DESCRIPTION
**Pull request by amartya-ws**
### Ticket Link

---

### Related Links

---

### Description
added the following features:
- reducer
- card component
- container
---

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### Live Link

---

- http://wednesday-react-template.s3-website.ap-south-1.amazonaws.com/<BRANCH_NAME>

### CODE REVIEW

File: app/components/ItuneCard/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    max-height: 14.5rem;
+    object-fit: cover;
+  }
+`;
+
+export function ItuneCard({ artistName, trackName, artworkUrl100 }) {
+  const { Meta } = Card;
+  return (
+    <div data-testid="itune-card">
+      <StyledCard cover={<StyledImg alt={'cover image'} src={artworkUrl100} data-testid="song-image" />}>
```
Comment (Anurag-Wednesday) [inline app/components/ItuneCard/index.js]: can we use a better alt tag

File: app/components/ItuneCard/tests/index.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  });
+
+  it('should contain 1 ItuneCard component', () => {
+    const { getAllByTestId } = renderWithIntl(<ItuneCard />);
+    expect(getAllByTestId('itune-card').length).toBe(1);
+  });
+
+  it('should render itune song details inside the card', () => {
+    let artistName = 'eminem';
+    let trackName = 'Till I Collapse (feat. Nate Dogg)';
```
Comment (Anurag-Wednesday) [inline app/components/ItuneCard/tests/index.test.js]: can we please create an object with the song details in describe and then access it using keys in the test

File: app/containers/SongContainer/index.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { compose } from 'redux';
+import { injectSaga } from 'redux-injectors';
+import { selectItuneData, selectArtistName } from './selectors';
+import { Input } from 'antd';
+import ItuneCard from '@app/components/ItuneCard';
+import { songContainerCreators } from './reducer';
+import For from '@app/components/For';
+import saga from './saga';
+import get from 'lodash/get';
+import styled from 'styled-components';
```
Comment (Anurag-Wednesday) [inline app/containers/SongContainer/index.js]: please fix the import order

File: app/containers/SongContainer/index.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+export function SongContainer({ artistName, ituneData, dispatchItuneSongs, dispatchClearSongs }) {
+  const changeHandler = (evt) => {
+    let searchTerm = evt.target.value;
+    if (searchTerm) {
+      dispatchItuneSongs(searchTerm);
+    } else {
+      dispatchClearSongs();
+    }
+  };
+
```
Comment (Anurag-Wednesday) [inline app/containers/SongContainer/index.js:57]: can we please add a debounce to the dispatch

File: app/containers/SongContainer/index.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          ParentComponent={CardWrapper}
+          renderItem={(item, index) => <ItuneCard key={index} {...item} />}
+        />
+      </>
+    );
+  };
+
+  return (
+    <Wrapper>
+      <h2>Get itune song details</h2>
```
Comment (Anurag-Wednesday) [inline app/containers/SongContainer/index.js]: Please use T component

File: app/components/ItuneCard/index.js (hunk 1 above)
Comment (amartya-ws) [reply app/components/ItuneCard/index.js]: done

File: app/containers/SongContainer/index.js (hunk 3 above)
Comment (amartya-ws) [reply app/containers/SongContainer/index.js]: done

File: app/components/ItuneCard/tests/index.test.js (hunk 2 above)
Comment (amartya-ws) [reply app/components/ItuneCard/tests/index.test.js]: done

File: app/containers/SongContainer/index.js (hunk 5 above)
Comment (amartya-ws) [reply app/containers/SongContainer/index.js]: done

File: app/containers/SongContainer/index.js (hunk 4 above)
Comment (amartya-ws) [reply app/containers/SongContainer/index.js:57]: done

**Review by Anurag-Wednesday — APPROVED** [review]

**Pull request merged**